class/UamUserGroup.php

```diff
@@ -135,39 +135,44 @@ public function save($blRemoveOldAssignments = true)
         $oDatabase = $this->getAccessHandler()->getUserAccessManager()->getDatabase();
         
         if ($this->_iId == null) {
-            $sInsertQuery = "
-                INSERT INTO " . DB_ACCESSGROUP . " (
-                    ID,
-                    groupname,
-                    groupdesc,
-                    read_access,
-                    write_access,
-                    ip_range
+            $oDatabase->insert(
+                DB_ACCESSGROUP,
+                array(
+                    'groupname'    => $this->_sGroupName,
+                    'groupdesc'    => $this->_sGroupDesc,
+                    'read_access'  => $this->_sReadAccess,
+                    'write_access' => $this->_sWriteAccess,
+                    'ip_range'     => $this->_sIpRange
+                ),
+                array(
+                    '%s',
+                    '%s',
+                    '%s',
+                    '%s',
+                    '%s',
                 )
-                VALUES (
-                    NULL,
-                    '{$this->_sGroupName}',
-                    '{$this->_sGroupDesc}',
-                    '{$this->_sReadAccess}',
-                    '{$this->_sWriteAccess}',
-                    '{$this->_sIpRange}'
-                )";
-
-            $oDatabase->query($sInsertQuery);
+            );
             
             $this->_iId = $oDatabase->insert_id;
         } else {
-            $sAccessGroupTable = DB_ACCESSGROUP;
-            $sUpdateQuery = "
-                UPDATE {$sAccessGroupTable} 
-                SET groupname = '{$this->_sGroupName}',
-                    groupdesc = '{$this->_sGroupDesc}',
-                    read_access = '{$this->_sReadAccess}',
-                    write_access = '{$this->_sWriteAccess}',
-                    ip_range = '{$this->_sIpRange}'
-                WHERE ID = {$this->_iId}";
-
-            $oDatabase->query($sUpdateQuery);
+            $oDatabase->update(
+                DB_ACCESSGROUP,
+                array(
+                    'groupname'    => $this->_sGroupName,
+                    'groupdesc'    => $this->_sGroupDesc,
+                    'read_access'  => $this->_sReadAccess,
+                    'write_access' => $this->_sWriteAccess,
+                    'ip_range'     => $this->_sIpRange
+                ),
+                array( 'ID' => $this->_iId),
+                array(
```

Comment:
Yes me too, but that is so many redundant code. I've seen that you can use '%s' instead of array('%s', '%s', '%s', '%s') if the vars are all from the same type. Maybe that is a good compromise.